lib/practice_exercises.rb

```diff
@@ -1,13 +1,29 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n)
+# Space Complexity: o(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  reverse_string = []
+  split_string = string.downcase.scan(/[a-z]/)
+
+  split_string.each_index{|i| reverse_string[i] =  split_string[split_string.length - 1 - i]}
+
+  return split_string == reverse_string
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n^2)
+# Space Complexity: o(n^2)
 def longest_prefix(strings)
```

Comment:
👍  Very clever solution.

The time and space complexities depend on the number of strings, n, and the length of the longest string, m.  So this would have O(nm).